File: history/0.1.2/tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/TransferObjectFactory.cs
using QiLeYuan.Tatala.proxy;

namespace QiLeYuan.Tatala.socket.to {

    public class TransferObjectFactory {

        private string connectionName;
        private bool longConnection;
        private DefaultProxy serverCallProxy;

        private string calleeClass;
        private bool compress;

        public TransferObjectFactory(string connectionName) {
            this.connectionName = connectionName;
        }

        public TransferObjectFactory(string connectionName, bool longConnection) {
            this.connectionName = connectionName;
            this.longConnection = longConnection;
        }

        public void registerServerCallProxy(DefaultProxy serverCallProxy) {
            this.serverCallProxy = serverCallProxy;
        }

        public void setCalleeClass(string calleeClass) {
            this.calleeClass = calleeClass;
        }

        public bool isCompress() {
            return compress;
        }

        public void setCompress(bool compress) {
            this.compress = compress;
        }

        public StandardTransferObject createTransferObject() {
            StandardTransferObject to = new StandardTransferObject();
            to.setConnectionName(connectionName);
            to.setLongConnection(longConnection);
            to.setServerCallProxy(serverCallProxy);

            if (calleeClass != null) {
                to.setCalleeClass(calleeClass);
            }
            to.setCompress(compress);
            return to;
        }

        public NewTransferObject createNewTransferObject() {
            NewTransferObject to = new NewTransferObject();
            to.setConnectionName(connectionName);
            to.setLongConnection(longConnection);
            to.setServerCallProxy(serverCallProxy);
            if (calleeClass != null) {
                to.setCalleeClass(calleeClass);
            }
            to.setCompress(compress);
            to.setNewVersion(true);
            return to;
        }
    }
}

File: tatala-java/src/main/java/com/qileyuan/tatala/socket/to/TransferObjectFactory.java
package com.qileyuan.tatala.socket.to;

import com.qileyuan.tatala.proxy.DefaultProxy;
import com.qileyuan.tatala.socket.client.SocketConnection;

/**
 * This class is transfer object factory class, which generates 
 * transfer object sent to different socket connection.
 * 
 * @author JimT
 *
 */
public class TransferObjectFactory {

	private DefaultProxy serverCallProxy;
	
	private String calleeClass;
	private String implClass;
	private boolean compress;
	
	private SocketConnection connection;

	public TransferObjectFactory(String ip, int port, int timeout) {
		this.connection = new SocketConnection(ip, port, timeout);
	}
	
	public TransferObjectFactory(String zkRegistryAddress, int timeout) {
		this.connection = new SocketConnection(zkRegistryAddress, timeout);
	}
	
	public void registerServerCallProxy(DefaultProxy serverCallProxy){
		this.serverCallProxy = serverCallProxy;
	}
	
	public void setCalleeClass(String calleeClass){
		this.calleeClass = calleeClass;
	}
	
	public void setCalleeClass(Class<?> calleeClass){
		this.calleeClass = calleeClass.getName();
	}
	
	public String getImplClass() {
		return implClass;
	}

	public void setImplClass(String implClass) {
		this.implClass = implClass;
	}
	
	public void setImplClass(Class<?> implClass) {
		this.implClass = implClass.getName();
	}

	public boolean isCompress() {
		return compress;
	}

	public void setCompress(boolean compress) {
		this.compress = compress;
	}
	
	public TransferObject createTransferObject(){
		TransferObject to = new OrderedTransferObject();
		to.setConnection(connection);
		to.setServerCallProxy(serverCallProxy);
		if(calleeClass != null){
			to.setCalleeClass(calleeClass);
		}
		to.setCompress(compress);
		return to;
	}
}
